### PR TITLE
fix: Remove auto-height from list widget's children

### LINKED ISSUE #18825: [Task]: Remove Auto Height from List widget's Children
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- [ ] Remove Auto height Property Control from List widget's Children
- [ ] Change Auto Height to Fixed when moving widget to List Widget
- [ ] Migrate existing list widget's Children to have Fixed Height

### PR DESCRIPTION
**Pull request by rahulramesha**
## Description

This PR contains code to remove auto height capability from List Widget's children. This includes,
- Changes to remove auto height control from property pane of list widget's Children
- Changes to convert list widget's children to fixed height when added to list widget
- Changes to Migrate existing list widget children to have fixed height

Fixes #18825

Media
https://www.loom.com/share/405060651fe04dad89560f2a8969c409

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual
- Jest
- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by rahulramesha** [comment]
@riodeuno please go through the PR and Let me know about your thoughts. There are abstraction leaks but not sure how to achieve this in another way.

File: app/client/src/widgets/ListWidget/index.ts [hunk 1]
```diff
@@ -375,18 +380,17 @@ export const CONFIG = {
...
                 message: `This widget cannot be used inside the list widget.`,
               };
             }
 
-            const template = {
-              ...get(parent, "template", {}),
-              [widget.widgetName]: widget,
-            };
-            parent.template = template;
+            // const template = {
```
Comment (riodeuno) [inline app/client/src/widgets/ListWidget/index.ts:389]: Let's remove this permanently.

**Review by riodeuno — DISMISSED** [review]
This needs a proper regression in List widget:
Scenarios to test: @appsmithorg/qa 
- Widgets within list widget have no dynamic height
- Widgets within list widget in existing applications have no dynamic height
- Widgets when moved out of the list widget have dynamic height in property pane
- Widgets when moved into the list widget have no dynamic height
- Widgets which have dynamic height outside the list widget when moved into the list widget should no longer have dynamic height
- Widgets when copied and pasted into the list widget no longer have dynamic height
- Widgets when cut and pasted into the list widget no longer have dynamic height
- Widgets which were not possible to be added to list widget cannot be pasted/moved into the list widget.

**Comment by kamakshibhat-appsmith** [comment]
@rahulramesha @riodeuno I cannot see relevant properties coming in the **content** section. It is just displaying a replica of **style**  in all the widgets
Screen Shot 2022-12-12 at 4 36 45 PM

**Comment by rahulramesha** [comment]
@kamakshibhat-appsmith This should be fixed now please check the latest dp once it is built

File: app/client/src/pages/Editor/PropertyPane/PropertyControl.tsx [hunk 2]
```diff
@@ -62,6 +62,7 @@ type Props = PropertyPaneControlConfig & {
   panel: IPanelProps;
   theme: EditorTheme;
   isSearchResult: boolean;
+  enhancements: EnhancementFns | undefined;
```
Comment (riodeuno) [inline app/client/src/pages/Editor/PropertyPane/PropertyControl.tsx:65]: If I understand this correctly, we're passing these to all the property controls from the generator. This means that the enhancements are independent of the property and apply to the widget itself.

**Review by riodeuno — DISMISSED** [review]

**Review by aswathkk — APPROVED** [review]

**Comment by kamakshibhat-appsmith** [comment]
Checked on the DP, This has been working fine . Moving to Done

**Review by riodeuno — APPROVED** [review]

**Pull request merged**